pyicloud/services/photos.py

```diff
@@ -474,7 +514,8 @@ def _list_query_gen(self, offset, list_type, direction, query_filter=None):
         return query
 
     def __unicode__(self):
-        return self.title
+        names = self.lineage + [self.title]
+        return '/'.join(names)
```

Comment:
I do not believe, this is a good logic here. In iCloud Photos, a folder name can contain a "/".

So either you need to escape the / character in foldernames (and possible also Album and Filenames) or you skip the whole thing.
Considering that none of the current services (ubuiquity, files and photos) support Unix-Styled foldernames/paths but instead do it via chaining of list items I'm thinking that maybe dropping this kind of feature completely is not the worst idea.

But let#s see what @Quentame thinks...